src/Shared/NodePacketTranslator.cs

```diff
@@ -341,7 +341,30 @@ public void TranslateDotNet<T>(ref T value)
                 BinaryFormatter formatter = new BinaryFormatter();
                 value = (T)formatter.Deserialize(_packetStream);
             }
+#else
+            public void TranslateBuildEventArgs(ref BuildEventArgs value)
+            {
+                var serializedBuildEventArgs = new SerializedBuildEventArgs();
+                serializedBuildEventArgs.Translate(this);
+                value = SerializedBuildEventArgs.ToEventArgs(serializedBuildEventArgs);
```

Comment:
I don't follow.  The signature of the method allows for the value to be written to the stream or read from it, and this is the implementation that reads from the stream.